src/http_client_async.zig

```diff
@@ -741,6 +771,30 @@ pub const HTTPThread = struct {
     }
 
     pub fn connect(this: *@This(), client: *HTTPClient, comptime is_ssl: bool) !NewHTTPContext(is_ssl).HTTPSocket {
+        if (comptime is_ssl) {
+            const needs_own_context = client.tls_props != null and client.tls_props.?.requires_custom_request_ctx;
+            if (needs_own_context) {
+                var requested_config = client.tls_props.?;
+                for (custom_ssl_context_map.keys()) |other_config| {
+                    if (requested_config.isSame(&other_config)) {
+                        // we freee the callers config since we have a existing one
+                        requested_config.deinit();
+                        client.tls_props = other_config;
+                        return try custom_ssl_context_map.get(other_config).?.connect(client, client.url.hostname, client.url.getPortAuto());
+                    }
+                }
+                // we need the config so dont free it
+                var custom_context = try bun.default_allocator.create(NewHTTPContext(is_ssl));
+                custom_context.initWithClientConfig(client) catch |err| {
+                    requested_config.deinit();
+                    client.tls_props = null;
+                    bun.default_allocator.destroy(custom_context);
+                    return err;
+                };
+                try custom_ssl_context_map.put(requested_config, custom_context);
```

Comment:
When do we free these SSL contexts after they're no longer in use?

Reply:
They don't get freed at all since im not entirely sur how to be sure its save to deallocate them and where, since the map lives independent of a request, if you have suggestions where il implement them.
At first i thought about implementing a max size limit for this map but then realised how can i be sure no request is using the context.